api/src/main/java/org/openmrs/module/commonreports/reports/AntenatalReportManager.java

```diff
@@ -0,0 +1,378 @@
+		                + "' and retired = 0) and order_type_id =(select order_type_id from order_type where uuid = '"
+		                + inizService.getValueFromKey("report.antenatal.drugOrder") + "')");
+		sqd.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		sqd.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		CompositionCohortDefinition ccd1 = new CompositionCohortDefinition();
+		ccd1.initializeFromElements(_prenatal, sqd, female);
+		antenatalRisks.addRow(MessageUtil.translate("commonreports.report.antenatalRisks.prenatalIron"), ccd1,
+		    parameterMappings);
+		
+		//Prenatal visit treated for Fe def (Same as above--> Prenatal visit + Fer Folate Co prescribed)
+		antenatalRisks.addRow(MessageUtil.translate("commonreports.report.antenatalRisks.prenatalIronDef"), ccd1,
+		    parameterMappings);
+		
+		//Mothers with a birth plan
+		NumericObsCohortDefinition scd = new NumericObsCohortDefinition();
+		scd.setGroupingConcept(inizService.getConceptFromKey("report.antenatal.estimatedGestationalAge"));
+		scd.setQuestion(inizService.getConceptFromKey("report.antenatal.numberOfWeeks"));
+		scd.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		scd.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		
+		antenatalRisks.addRow(MessageUtil.translate("commonreports.report.antenatalRisks.motherBirthPlan"), scd,
+		    parameterMappings);
+		
+		//Prenatal visit + malaria test positive + chloroquine co prescribed 
+		CodedObsCohortDefinition malaria = new CodedObsCohortDefinition();
+		malaria.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		malaria.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		malaria.setOperator(SetComparator.IN);
+		
+		malaria.setQuestion(inizService.getConceptFromKey("report.antenatal.malaria"));
+		
+		List<Concept> malariaPositiveConceptSet = new ArrayList<Concept>();
+		malariaPositiveConceptSet.add(inizService.getConceptFromKey("report.antenatal.positive"));
+		malariaPositiveConceptSet.add(inizService.getConceptFromKey("report.antenatal.onePlus"));
+		malariaPositiveConceptSet.add(inizService.getConceptFromKey("report.antenatal.twoPlus"));
+		malariaPositiveConceptSet.add(inizService.getConceptFromKey("report.antenatal.threePlus"));
+		malariaPositiveConceptSet.add(inizService.getConceptFromKey("report.antenatal.fourPlus"));
+		
+		malaria.setValueList(malariaPositiveConceptSet);
+		
+		SqlCohortDefinition sql = new SqlCohortDefinition(
+		        "select patient_id from orders where concept_id=(select DISTINCT concept_id from concept where uuid = '"
+		                + inizService.getValueFromKey("report.antenatal.chloroquine")
+		                + "' and retired = 0) and order_type_id =(select order_type_id from order_type where uuid = '"
+		                + inizService.getValueFromKey("report.antenatal.drugOrder") + "')");
```

Comment:
If you do not add date parameters, this query will collect all possible orders. Let's therefore add something like `... AND date_created BETWEEN :onOrAfter AND : onOrBefore `

Reply:
Isn't this handled in the following line 247

Reply:
Same here...

Reply:
Done